plugin/cache/handler.go

```diff
@@ -27,18 +27,33 @@ func (c *Cache) ServeDNS(ctx context.Context, w dns.ResponseWriter, r *dns.Msg)
 	server := metrics.WithServer(ctx)
 
 	i, found := c.get(now, state, server)
-	if i != nil && found {
-		resp := i.toMsg(r, now)
-		w.WriteMsg(resp)
-
-		if c.shouldPrefetch(i, now) {
-			go c.doPrefetch(ctx, state, server, i, now)
+	if i == nil || !found {
+		if c.serveExpired {
+			i, found = c.get(now.Add(-c.expiredUpTo), state, server)
```

Comment:
why do we need to reget the item; we've already looked it up. It's it more natural to have `c.get` return an boolean that includes that's it expired?
Or maybe a new function getExpired, but we control the intree code so we can upgrade all callers.

Reply:
You are right.
I added a getItem to unconditionally get an item from the cache. PTAL.